1. Lectura de archivos
- Importa pandas y numpy siguiendo las convenciones.
- Lee la base de datosworldcup2014.csv y asígnela aun objetodf.
- Solicita las primeras 5 observaciones conhead.

In [1]:
import pandas as pd
import numpy as np

df = pd.read_csv('worldcup2014.csv')

df.head(5)

team     continent group  group_pos  cantidad_juegos  \
0       Brazil  southamerica     A          1                3   
1       Mexico  northamerica     A          2                3   
2      Croatia        europe     A          3                3   
3     Cameroon        africa     A          4                3   
4  Netherlands        europe     B          1                3   

   juegos_ganados  juegos_empatados  juegos_perdidos  goles_favor  \
0               2                 1                0            7   
1               2                 1                0            4   
2               1                 0                2            6   
3               0                 0                3            1   
4               3                 0                0           10   

   goles_contra goles_diferencia  puntos  clasificado  
0             2                5       7            1  
1             1                3       7            1  
2             6                0       3            0  
3             9               −8       0            0  
4             3                7       9            1

2. Frecuencia de continent: utilizando value_counts responde lo siguiente:
- ¿Cuál es el continente con una mayor presencia en la muestra?
- ¿Cuál es la probabilidad de elegir un equipo asiático al azar?
- ¿Cuál es la probabilidad de elegir un equipo africano al azar?

In [2]:
df['continent'].value_counts('%')

europe          0.40625
southamerica    0.18750
africa          0.15625
northamerica    0.12500
asia            0.12500
Name: continent, dtype: float64

- El continente con mayor presencia es Europa, con un 40.625% de los equipos siendo de este continente

- La probabilidad de elegir un equipo asiatico es del 12.5%

- La probabilidad de elegir un equipo africano es del 15.625%

3. Generación de un subconjunto de datos: por cada continente, genere un nuevo
objeto que almacene sólo las observaciones del continente.
Tip: Para ello puede utilizar la siguiente sintaxis: df[df['variable'] ==
condicion].


In [9]:
europe = df[df['continent'] == 'europe']
southamerica = df[df['continent'] == 'southamerica']
africa = df[df['continent'] == 'africa']
northamerica = df[df['continent'] == 'northamerica']
asia = df[df['continent'] == 'asia']

4. Probabilidad de clasificación a la siguiente ronda: calcula la probabilidad de que un
país pase a la siguiente ronda (clasifique) por continente.
- ¿Cuál es la probabilidad de que un país asiático pase a la siguiente ronda?
- ¿Cuáles son los dos continentes con la mayor probabilidad de clasificar?
- ¿Cuál es la probabilidad de que un país europeo no clasifique?

Tip: El problema puede ser resuelto de varias formas:
- La manera más intuitiva corresponde a calcular la probabilidad en cada
subconjunto creado en el problema 3.
- Una forma alternativa es copiar todos los subsets creados en el problema 3
en una lista e iterar sobre dicha lista.
- Una forma más elegante consiste en utilizar una operación de agrupación por
continente, y luego calcular la probabilidad.


In [6]:
def calc_prob_clasif(df):
    probabilidad = df['clasificado'].value_counts('%')
    return probabilidad

In [29]:
df.groupby('continent')['clasificado'].value_counts('%')

continent     clasificado
africa        0              0.600000
              1              0.400000
asia          0              1.000000
europe        0              0.538462
              1              0.461538
northamerica  1              0.750000
              0              0.250000
southamerica  1              0.833333
              0              0.166667
Name: clasificado, dtype: float64

- La probabilidad de que un pais asiatico pase a la siguiente ronda es del 0%
- Sudamerica y Norteamerica son los dos paises con mayor probabilidad de clasificar
- La probabilidad de que un pais Europeo NO clasifique es del 53.8462%

5. Probabilidad de victoria y derrota: se pide utilizar value_counts para calcular la
probabilidad de que un continente contenga países con o sin victorias.
Para ello crea una columna adicional llamada 'ha_ganado' que contenga un 0 si es
que un país no tiene juegos ganados y un 1 si es que ha ganado al menos un juego.


In [31]:
df['ha_ganado'] = np.where(df['juegos_ganados'] > 0, 1, 0)
df.groupby('continent')['ha_ganado'].value_counts('%')

continent     ha_ganado
africa        1            0.600000
              0            0.400000
asia          0            1.000000
europe        1            0.846154
              0            0.153846
northamerica  1            0.750000
              0            0.250000
southamerica  1            1.000000
Name: ha_ganado, dtype: float64

- Con base en el resultado del procedimiento recién descrito, responda lo
siguiente:
1. ¿Qué continente tiene la mayor probabilidad de tener países con al menos 1
victoria?
2. ¿Qué continente tiene un nivel similar entre países que tienen o no tienen
victorias?

- El país que tiene la mayor probabilidad de paises que tienen al menos 1 victoria es Europa
- El país que tiene un nivel similar entre países que tienen o no tienen victorias es Africa

- Reformule el código para juegos perdidos y analice los continentes con mayor
probabilidad de tener países sin juegos perdidos.

In [33]:
df['ha_perdido'] = np.where(df['juegos_perdidos'] > 0, 1, 0)
df.groupby('continent')['ha_perdido'].value_counts('%')

continent     ha_perdido
africa        1             1.000000
asia          1             1.000000
europe        1             0.692308
              0             0.307692
northamerica  0             0.500000
              1             0.500000
southamerica  0             0.500000
              1             0.500000
Name: ha_perdido, dtype: float64

- Los continentes que tienen una mayor probabilidad de tener paises sin juegos perdidos son Norteamérica y Sudamérica